code/ui/blocks/src/components/TableOfContents.tsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect } from 'react';
+import type { FC, ReactElement } from 'react';
+import { styled } from '@storybook/theming';
+import tocbot from 'tocbot';
+
+export interface TocParameters {
+  /** CSS selector for the container to search for headings. */
+  contentsSelector?: string;
+
+  /**
+   * When true, hide the TOC. We still show the empty container
+   * (as opposed to showing nothing at all) because it affects the
+   * page layout and we want to preserve the layout across pages.
+   */
+  disable?: boolean;
+
+  /** CSS selector to match headings to list in the TOC. */
+  headingSelector?: string;
+
+  /** Headings that match the ignoreSelector will be skipped. */
+  ignoreSelector?: string;
+
+  /** Custom title ReactElement or string to display above the TOC. */
+  title?: ReactElement | string | null;
+
+  /**
+   * TocBot options, not guaranteed to be available in future versions.
+   * See [tocbot docs](https://tscanlin.github.io/tocbot/#usage)
```

Comment:
There's JSDoc syntax for this which TS and VSCode hover supports.

https://jsdoc.app/tags-see.html

```suggestion
   * @see tocbot docs {@link https://tscanlin.github.io/tocbot/#usage}
```